docs/madr/decisions/070-resource-identifier.md

```diff
@@ -0,0 +1,363 @@
+| FilterChain | when `<match-value> == <protocol>` -> `meshpassthrough_<protocol>_<port>`<br>when `<match-value> != <protocol>` -> `meshpassthrough_<protocol>_<match-value>_<port>`<br>when port is 0 we put `*`, i.e. `meshpassthrough_http_*`<br>`<match-value>` is defined above                                                                                                                                                               | –                    |
+
+##### Internal
+
+|                  | Name                                                                                                                                                                 | Correlated Resources |
+|------------------|----------------------------------------------------------------------------------------------------------------------------------------------------------------------|----------------------|
+| Listener         | `kuma:dns`<br>`kuma:envoy:admin`<br>`kuma:metrics:prometheus`<br>`_kuma:metrics:opentelemetry:<backendName>`                                                         | –                    |
+| VirtualHost      | listener's name                                                                                                                                                      | –                    |
+| Internal Cluster | `kuma:readiness`<br>`kuma:envoy:admin`<br>`kuma:metrics:hijacker`<br>`_kuma:metrics:opentelemetry:<backendName>`<br>`tracing:`<br>`access_log_sink`<br>`ads_cluster` | –                    |
+
+#### Envoy stats
+
+There are 2 fields to specify alternative name for emitting stats – `stat_prefix`, `alt_stat_name`.
+Field `alt_stat_name` is used only for Cluster, and it's automatically converts all `:` to `_` by Envoy when emitting statistics.
+Field `stat_prefix` is used for HTTPConnectionManager, TCPProxy, KafkaBroker (network filter) and RBAC (network filter).
+Unlike `alt_stat_name` the field `stat_prefix` doesn't have any restrictions from Envoy.
+Kuma sanitizes both fields using `SanitizeMetric` function before setting them on Envoy's resource:
+
+```go
+var illegalChars = regexp.MustCompile(`[^a-zA-Z_\-0-9]`)
+
+// We need to sanitize metrics in order to  not break statsd and prometheus format.
+// StatsD only allow [a-zA-Z_\-0-9.] characters, everything else is removed
+// Extra dots breaks many regexes that converts statsd metric to prometheus one with tags
+func SanitizeMetric(metric string) string {
+	return illegalChars.ReplaceAllString(metric, "_")
+}
+```
+
+I can't find explanation why `alt_stat_name` and `stat_prefix` are treated differently by Envoy. 
```

Comment:
A use case could be that `stat_prefix` and `alt_stat_name` parameters can be used to consolidate statistics from multiple similar entities under a single name.

For example, if you have separate IPv4 and IPv6 clusters named `my-cluster:v4` and `my-cluster:v6`, you can use `alt_stat_name: my-cluster` for both clusters to aggregate their statistics under the common name `my-cluster` when exporting metrics. This allows you to track related entities as a unified group while maintaining distinct names for configuration purposes.

WDYT?

Reply:
Ah no, I don't question why there is an actual name and stats name, I just don't understand why `stat_prefix` behaves differently not the same way as `alt_stat_name`.